src/nanosvg.h

```diff
@@ -809,17 +816,28 @@ static NSVGgradientData* nsvg__findGradientData(NSVGparser* p, const char* id)
 	return NULL;
 }
 
-static NSVGgradient* nsvg__createGradient(NSVGparser* p, const char* id, const float* localBounds, char* paintType)
+static NSVGgradientLink* nsvg__createGradientLink(const char* id, const float *xform)
+{
+	NSVGgradientLink* grad = (NSVGgradientLink*)malloc(sizeof(NSVGgradientLink));
+	if (grad == NULL) return NULL;
+	strncpy(grad->id, id, 63);
+	grad->id[63] = '\0';
+	memcpy(grad->xform, xform, sizeof(float)*6);
+	return grad;
+}
+
+static void nsvg__getLocalBounds(float* bounds, NSVGshape *shape, float* xform);
+
+static NSVGgradient* nsvg__createGradient(NSVGparser* p, NSVGshape* shape, NSVGgradientLink* link, char* paintType)
 {
-	NSVGattrib* attr = nsvg__getAttr(p);
 	NSVGgradientData* data = NULL;
 	NSVGgradientData* ref = NULL;
 	NSVGgradientStop* stops = NULL;
 	NSVGgradient* grad;
 	float ox, oy, sw, sh, sl;
 	int nstops = 0;
 
-	data = nsvg__findGradientData(p, id);
+	data = nsvg__findGradientData(p, link->id);
```

Comment:
here what if link==NULL?

Reply:
I changed `nsvg__assignGradients`to gracefully handle `type == NSVG_PAINT_GRADIENT_LINK && gradientLink == NULL` (which, with the modification above, should never happen anyway, so we should be safe now).